tests/integration/fixtures/processing.py

```diff
@@ -270,89 +272,123 @@ def assert_rate_limited(
             assert count == quantity
 
 
+def consumer_fixture(f):
+    consumer: list[Optional[ConsumerBase]] = [None]
+
+    @functools.wraps(f)
+    def inner(*args, **kwargs):
+        consumer[0] = f(*args, **kwargs)
+        return consumer[0]
+
+    yield inner
+
+    if consumer[0] is not None:
+        consumer[0].assert_empty()
+
+
 @pytest.fixture
 def events_consumer(kafka_consumer):
-    return lambda timeout=None: EventsConsumer(
-        timeout=timeout, *kafka_consumer("events")
+    yield from consumer_fixture(
+        lambda timeout=None: EventsConsumer(timeout=timeout, *kafka_consumer("events"))
```

Comment:
Can we make this `yield from consumer_fixture(EventsConsumer, "events")`? I.e. move the lambda and the call to `kafka_consumer` into`consumer_fixture`.